common/src/main/java/software/tnb/common/utils/WaitUtils.java

```diff
@@ -84,14 +90,20 @@ public static void waitFor(BooleanSupplier resourceCheck, int retries, long wait
      * @param logMessage log message that will be printed out before waiting
      * @throws FailureConditionMetException when the fail condition is true
      */
-    public static void waitFor(BooleanSupplier check, BooleanSupplier fail, long timeout, String logMessage) throws FailureConditionMetException {
+    public static void waitFor(BooleanSupplier check, BooleanSupplier fail, long timeout, String logMessage, Supplier<FailureCauseException> failureCause) throws FailureConditionMetException {
         LOG.info(logMessage);
         Instant start = Instant.now();
         while (true) {
             if (check.getAsBoolean()) {
                 break;
             } else if (fail.getAsBoolean()) {
-                throw new FailureConditionMetException("Specified fail condition met");
+                String exceptionMessage = "Specified fail condition met";
```

Comment:
maybe wrap into `if (TestConfiguration.streamLogs())` as well?

Reply:
Good point, I wrapped it in the App class where the method is called